### PR TITLE
Support custom baseurl and jekyll serve. Update Gems.

### PR DESCRIPTION
**Pull request by haxpor**
This updated to use newer version of Gems-dependencies.
There are still warning and deprecated shouting out when doing "bundle
exec jekyll serve" but it works at the end for this commit for now.

To get the path of asset instead of using normal way of `{% javascript
app.js %}`, we use `<script type="text/javascript" src="{{ site.baseurl
}}{% asset_path app.js %}"></script>` instead as we need to prepend
site.baseurl.

baseurl and url are added to _config.yml.

So in short, when in development just hit `bundle exec jekyll serve`. It
will provide live compile, and you can see immediate update on
http://localhost:4000.

### CODE REVIEW

**Comment by haxpor** [comment]
I found out now that those warning and deprecated on console is due to the project uses `jekyll-assets`'s version `0.2.1`. But due to newer version cut out erb, I might take a look to update it later but for now it do its work.

**Comment by ixti** [comment]
I think it's better to use new version of jekyll-assets (2.0+) with latest jekyll for new projects. This repo was created a while ago when I just released first version of jekyll-assets.

**Comment by ixti** [comment]
After re-reading your comment, and taking a look on what happened to jekyll-assets I got what you mean. And I think jekyll-assets will never become available on GH pages, thus stripping of ERB was a bad move. It should  be pretty simple to bring it back though for your own website, after all it's just Sprocket in a wrapper ;))

I'll take a look over the weekend.

**Comment by haxpor** [comment]
Hi :) Is there any update on this? I'd like to be able to remove all those warnings.

**Comment by ixti** [comment]
Oh sorry. Got crazy days. Will do my best to take care of this during this week, max over upcoming weekend.

**Comment by haxpor** [comment]
Hey thank you!.
PS: Sorry if my comment sounds pressuring, I don't mean that.

**Comment by ixti** [comment]
@haxpor All good. I really simply forgot about this PR with all the stuff... :D So thanks for a reminder.

**Comment by ixti** [comment]
Sorry took me a bit longer to get to this. Please take a look at my branch, there's only one commit that is relevant: https://github.com/ixti/jekyll-assets-demo/commit/d5dc3300619e5829aab55f1b4b8324cc1356a623

It updates this repo to use latest jekyll, jekyll-assets and configures latter one to support ERB again :D

**Comment by ixti** [comment]
I believe there's no point in support of the jekyll-assets version that is not maintained anymore. As I have updated deps to latest versions in this repo I think this PR might be closed.

**Comment by haxpor** [comment]
@ixti hey, I tested and integrated with my site. Works great! No more warning, all green and clean. Thanks again!

**Comment by ixti** [comment]
@haxpor you are welcome! :D
